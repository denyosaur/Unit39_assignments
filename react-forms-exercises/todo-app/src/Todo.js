import React from "react";

const Todo = ({ task, deleteTodo }) => {
    const handleDelete = (evt) => {
        evt.preventDefault();
        deleteTodo();
    }

    return <div className="Todo">
        <span className="Todo-task">{task}</span>
        <button className="Todo-delete" onClick={handleDelete}>X</button>
    </div>

}

export default Todo;