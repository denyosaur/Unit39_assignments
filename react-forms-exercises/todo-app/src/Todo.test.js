import { render, fireEvent } from "@testing-library/react";
import Todo from "./Todo";

//smoke test
test("Todo should render", () => {
    render(<Todo />)
})

//snapshot test
test("Todo should render", () => {
    const { asFragment } = render(<Todo />)
    expect(asFragment()).toMatchSnapshot();
})

//Delete button is present
test("Delete Todo", () => {
    const { getByText } = render(<Todo task="clean stuff" />);
    const btn = getByText("X");
    expect(btn).toBeInTheDocument();
})