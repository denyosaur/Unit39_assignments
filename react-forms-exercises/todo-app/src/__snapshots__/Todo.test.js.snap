// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo should render 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <span
      class="Todo-task"
    />
    <button
      class="Todo-delete"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
